controllers/component_build_controller_unit_test.go

```diff
@@ -185,6 +188,128 @@ func TestWriteBuildStatus(t *testing.T) {
 	}
 }
 
+func TestGenerateInitialPipelineRunForComponentDevfileError(t *testing.T) {
+	component := &appstudiov1alpha1.Component{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-component",
+			Namespace: "my-namespace",
+			Annotations: map[string]string{
+				"skip-initial-checks":            "true",
+				gitops.GitProviderAnnotationName: "github",
+			},
+		},
+		Spec: appstudiov1alpha1.ComponentSpec{
+			Application:    "my-application",
+			ContainerImage: "registry.io/username/image:tag",
+			Source: appstudiov1alpha1.ComponentSource{
+				ComponentSourceUnion: appstudiov1alpha1.ComponentSourceUnion{
+					GitSource: &appstudiov1alpha1.GitSource{
+						URL:      "https://githost.com/user/repo.git",
+						Revision: "custom-branch",
+					},
+				},
+			},
+		},
+		Status: appstudiov1alpha1.ComponentStatus{
+			Devfile: "wrong",
+		},
+	}
+	pipelineRef := &tektonapi.PipelineRef{
+		Name:   "pipeline-name",
+		Bundle: "pipeline-bundle",
+	}
+	additionalParams := []tektonapi.Param{
+		{Name: "revision", Value: tektonapi.ArrayOrString{Type: "string", StringVal: "2378a064bf6b66a8ffc650ad88d404cca24ade29"}},
+		{Name: "rebuild", Value: tektonapi.ArrayOrString{Type: "string", StringVal: "true"}},
+	}
+	commitSha := "26239c94569cea79b32bce32f12c8abd8bbd0fd7"
+
+	pRunGitInfo := &buildGitInfo{
+		gitSourceSha:              commitSha,
+		browseRepositoryAtShaLink: "https://githost.com/user/repo?rev=" + commitSha,
+	}
+
+	_, err := generatePipelineRunForComponent(component, pipelineRef, additionalParams, pRunGitInfo)
+	if err == nil {
+		t.Error("generateInitialPipelineRunForComponentDevfileError(): Didn't return error")
+	} else {
+		gotError := strings.Contains(err.Error(), "failed to populateAndParseDevfile: failed to decode devfile json")
+		if gotError != true {
+			t.Errorf("generateInitialPipelineRunForComponentDevfileError(): Got incorrect error: '%v'", err)
+		}
+	}
```

Comment:
How about use `assert.ErrorContains` from package gotest.tools to simplify this if-else?

Reply:
changed